### ISSUE TITLE
data.table syntax

### ISSUE THREAD

**Issue by gsee** [trestletech/plumber#48]
If data.table syntax is used in a file, `plumb("myfile.R")` will fail.

here is an example "myfile.R"

```
library(data.table)
x <- as.data.table(BOD)
x[, demand]

#* @get /test
function() x
```

Now, when I try to `plumb` it I get an error

```
> p <- plumb("myfile.R")
Error in `[.data.frame`(x, i, j) : object 'demand' not found
```

**Comment by raubreywhite** [comment]
I have this problem too. Is there any news on it?

**Comment by ablazko** [comment]
@trestletech Hi Jeff, congrats for this package. It's really helpful!

But as the others mentioned, there is a bug in plumber when data.table syntax is used in the code. Here is a reproducible example:

The test is pretty simple. The user enter a numeric input and get 5 group names as a response.

### myfile.R
```r
# USING data.table syntax
#* @get /dt_apiTC
dt_apiTC <- function(idc=0)
{
    idc = as.numeric(idc)
    if (is.null(idc) | idc==0) { return("Invalid input parameter!") }
    
    # Create a table (using data.table)
    data_0 = data.table(cgroup = 1:5)
    data_0[, ":=" (cgroup = paste("Group", data_0$cgroup * idc))]

    # Create group names
    var_groups = data_0[,unique(cgroup)]
    
    return(var_groups)
}


# USING data.frame syntax
#* @get /df_apiTC
df_apiTC <- function(idc=0)
{
    idc = as.numeric(idc)
    if (is.null(idc) | idc==0) { return("Invalid input parameter!") }
    
    # Create a table (using data.frame)
    data_0 = data.frame(cgroup = 1:5)
    data_0$cgroup = paste("Group", data_0$cgroup * idc)

    # Create group names
    var_groups = unique(data_0$cgroup)
    
    return(var_groups)
}
```

### Initiate plumber
```r
library(data.table)
library(plumber)
r <- plumb("~/Documents/myfile.R")
r$run(port=8000)
```

### Results

#### data.table call and results
```r
# Called URL: "http://localhost:8000/dt_apiTC?idc=10"

# Results
{"error":["500 - Internal server error"],"message":["Check that is.data.table(DT) == TRUE. Otherwise, := and `:=`(...) are defined for use in j, once only and in particular ways. See help(\":=\")."]}
```

#### data.frame call and results
```r
# Called URL: "http://localhost:8000/df_apiTC?idc=10"

# Results
["Group 10","Group 20","Group 30","Group 40","Group 50"]
```


## Environment tested
```r
platform       x86_64-apple-darwin13.4.0   
arch           x86_64                      
os             darwin13.4.0                
system         x86_64, darwin13.4.0        
status                                     
major          3                           
minor          2.3                         
year           2015                        
month          12                          
day            10                          
svn rev        69752                       
language       R                           
version.string R version 3.2.3 (2015-12-10)
nickname       Wooden Christmas-Tree

‘plumber’ version 0.3.1
```

**Comment by trestletech** [comment]
@ablazko awesome comment. Thank you for taking the time to organize that, it made it much easier to confirm that this was fixed. You're allowed to open issues on my repos anytime ;)

This should be resolved as of PR #71, but please let me know if you find otherwise.

**Comment by ablazko** [comment]
@trestletech I appreciate your feedback, and I'm glad to help. :)
I'll update this (awesome!) package from CRAN (when available) and run these and others tests again. Hopefully I won't find any other issues. Congrats one more time for this amazing job! ;-)
